src/java/main/org/apache/zookeeper/ZooKeeper.java

```diff
@@ -1436,7 +1436,7 @@ public String create(final String path, byte data[], List<ACL> acl,
         request.setData(data);
         request.setFlags(createMode.toFlag());
         request.setPath(serverPath);
-        if (acl != null && acl.size() == 0) {
+        if (acl == null || acl.isEmpty() || acl.contains(null)) {
```

Comment:
Thanks @jainbhupendra24 for the details. We can leave these functions as it is. It is trivial problem we can ignore over the backward compatibility.